Installation issue: umpire

In #19411 `camp` was added as a dependency of `umpire`, but the source checksum of `camp` was wrong, so the `umpire` build didn't get a chance to run.  Then #19543 fixed that problem so that `camp` built, but now the `umpire` build fails during the configuration phase with a lot errors like the following:

```
  >> 66     CMake Error at /home/software/spack_path_placeholder/spack_path_pla
            ceholder/spack_path_placeholder/spack_path_placeholder/spack_path_p
            laceholder/spack_path_placeholder/spack_path_placeholder/spack_path
            _placeholder/spack_path_placeholder/spack_path_placeholder/spack_pa
            th_placeholder/spack_path_placeholder/spack_path_placeholder/spack_
            path_placeholder/spack_path_placeholder/spack_path_placeholder/spac
            k_path_placeholder/spack_path_placeholder/spack_path_placeholder/sp
            ack_path_placeholder/spack_path_placeholder/spack_path_placeholder/
            spack_/linux-ubuntu18.04-x86_64/gcc-7.3.0/blt-0.3.6-gn3xknsnfg5qkpn
            vkzcmdrlivwiudzm6/cmake/BLTPrivateMacros.cmake:338 (target_link_lib
            raries):
     67       Error evaluating generator expression:
     68
     69         $<LINK_LANGUAGE:Fortran>
     70
     71       Expression did not evaluate to a known generator expression
     72     Call Stack (most recent call first):
     73       /home/software/spack_path_placeholder/spack_path_placeholder/spac
            k_path_placeholder/spack_path_placeholder/spack_path_placeholder/sp
            ack_path_placeholder/spack_path_placeholder/spack_path_placeholder/
            spack_path_placeholder/spack_path_placeholder/spack_path_placeholde
            r/spack_path_placeholder/spack_path_placeholder/spack_path_placehol
            der/spack_path_placeholder/spack_path_placeholder/spack_path_placeh
            older/spack_path_placeholder/spack_path_placeholder/spack_path_plac
            eholder/spack_path_placeholder/spack_path_placeholder/spack_/linux-
            ubuntu18.04-x86_64/gcc-7.3.0/blt-0.3.6-gn3xknsnfg5qkpnvkzcmdrlivwiu
            dzm6/cmake/BLTMacros.cmake:604 (blt_setup_target)
     74       examples/CMakeLists.txt:70 (blt_add_executable)
``` 

### Steps to reproduce the issue

```console
$ spack install umpire
```

The above command currently picks up `umpire@4.1.2`.

### Information on your system

This happened on `linux-ubuntu18.04`, attempting to build for `x86_64` target architecture, using a system-provided `gcc@7.3.0`.  The version of `spack` was:

```
$ spack --version
 0.15.4-1688-9b809e01b
```

### Additional information

These builds are running as part of an automated gitlab build pipeline, I have attached the pipeline log as well as the spack build output below.

[pipeline_log.txt](https://github.com/spack/spack/files/5453182/pipeline_log.28.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/5453184/spack-build-out.12.txt)

--- a/var/spack/repos/builtin/packages/umpire/cmake_version_check.patch
+++ b/var/spack/repos/builtin/packages/umpire/cmake_version_check.patch
@@ -1,0 +1,13 @@
+diff --git a/src/umpire/CMakeLists.txt b/src/umpire/CMakeLists.txt
+index 18518f13..c2e45e03 100644
+--- a/src/umpire/CMakeLists.txt
++++ b/src/umpire/CMakeLists.txt
+@@ -105,7 +105,7 @@ blt_add_library(
+   DEPENDS_ON ${umpire_depends}
+   DEFINES ${umpire_defines})
+ 
+-if (NOT WIN32 AND CMAKE_VERSION VERSION_GREATER 3.17)
++if (NOT WIN32 AND NOT ${CMAKE_VERSION} VERSION_LESS 3.18)
+   target_link_options(
+     umpire INTERFACE
+     $<BUILD_INTERFACE:$<$<NOT:$<LINK_LANGUAGE:Fortran>>:-rdynamic>>

--- a/var/spack/repos/builtin/packages/umpire/package.py
+++ b/var/spack/repos/builtin/packages/umpire/package.py
@@ -14,6 +14,8 @@ class Umpire(CMakePackage, CudaPackage):
     homepage = 'https://github.com/LLNL/Umpire'
     git      = 'https://github.com/LLNL/Umpire.git'
 
+    maintainers = ['davidbeckingsale']
+
     version('develop', branch='develop', submodules='True')
     version('main', branch='main', submodules='True')
     version('4.1.2', tag='v4.1.2', submodules='True')

--- a/var/spack/repos/builtin/packages/umpire/package.py
+++ b/var/spack/repos/builtin/packages/umpire/package.py
@@ -42,6 +42,7 @@ class Umpire(CMakePackage, CudaPackage):
     version('0.1.3', tag='v0.1.3', submodules='True')
 
     patch('camp_target_umpire_3.0.0.patch', when='@3.0.0')
+    patch('cmake_version_check.patch', when='@4.1.0:')
 
     variant('fortran', default=False, description='Build C/Fortran API')
     variant('c', default=True, description='Build C API')

--- a/var/spack/repos/builtin/packages/umpire/package.py
+++ b/var/spack/repos/builtin/packages/umpire/package.py
@@ -42,7 +42,7 @@ class Umpire(CMakePackage, CudaPackage):
     version('0.1.3', tag='v0.1.3', submodules='True')
 
     patch('camp_target_umpire_3.0.0.patch', when='@3.0.0')
-    patch('cmake_version_check.patch', when='@4.1.0:')
+    patch('cmake_version_check.patch', when='@4.1.0:main')
 
     variant('fortran', default=False, description='Build C/Fortran API')
     variant('c', default=True, description='Build C API')

